Firebase/Storage/FIRStorageDeleteTask.m

```diff
@@ -51,20 +51,21 @@ - (void)enqueue {
 
   fetcher.comment = @"DeleteTask";
 
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Warc-retain-cycles"
+  __weak FIRStorageDeleteTask *weakSelf = self;
   _fetcherCompletion = ^(NSData *_Nullable data, NSError *_Nullable error) {
-    if (!self.error) {
-      self.error = [FIRStorageErrors errorWithServerError:error reference:self.reference];
-    }
-    if (callback) {
-      callback(self.error);
+    __strong FIRStorageDeleteTask *strongSelf = weakSelf;
+    if (strongSelf != nil) {
```

Comment:
Nit: I don't feel strongly, but rest of the code mostly uses `if(var)` for nil checks.